package cn.mcmod.sakura.item.enums;

import net.minecraft.world.effect.MobEffectInstance;
import net.minecraft.world.effect.MobEffects;

public enum SakuraAlcoholSet {
    GLASS_BEER("glass_beer",
            new MobEffectInstance(MobEffects.SATURATION, 200, 0)),
    GLASS_DOBUROKU("glass_doburoku",
            new MobEffectInstance(MobEffects.DAMAGE_RESISTANCE, 200, 0)),
    GLASS_SAKE("glass_sake",
            new MobEffectInstance(MobEffects.DAMAGE_RESISTANCE, 400, 0),
            new MobEffectInstance(MobEffects.REGENERATION, 100, 0)),
    GLASS_SHOUCHU("glass_shouchu",
            new MobEffectInstance(MobEffects.DAMAGE_RESISTANCE, 600, 0),
            new MobEffectInstance(MobEffects.REGENERATION, 200, 0)),
    GLASS_RED_WINE("glass_red_wine",
            new MobEffectInstance(MobEffects.NIGHT_VISION, 400, 0)),
    GLASS_WHITE_WINE("glass_white_wine",
            new MobEffectInstance(MobEffects.NIGHT_VISION, 400, 0)),
    GLASS_CHAMPAGNE("glass_champagne",
            new MobEffectInstance(MobEffects.NIGHT_VISION, 600, 1)),
    GLASS_RUM("glass_rum",
            new MobEffectInstance(MobEffects.WATER_BREATHING, 400, 0)),
    GLASS_VODKA("glass_vodka",
            new MobEffectInstance(MobEffects.DAMAGE_BOOST, 400, 0)),
    GLASS_WHISKEY("glass_whiskey",
            new MobEffectInstance(MobEffects.DIG_SPEED, 400, 0)),
    GLASS_BRANDY("glass_brandy",
            new MobEffectInstance(MobEffects.DIG_SPEED, 400, 0)),
    GLASS_GIN("glass_gin",
            new MobEffectInstance(MobEffects.DIG_SPEED, 400, 0)),
    GLASS_TEQUILA("glass_tequila",
            new MobEffectInstance(MobEffects.FIRE_RESISTANCE, 400, 0)),
    GLASS_LIQUEUR("glass_liqueur",
            new MobEffectInstance(MobEffects.JUMP, 400, 0)),
    GLASS_COCOA_LIQUEUR("glass_cocoa_liqueur",
            new MobEffectInstance(MobEffects.JUMP, 400, 0));

    private final String name;
    private final MobEffectInstance[] effects;

    SakuraAlcoholSet(String name, MobEffectInstance... effects) {
        this.name = name;
        this.effects = effects;
    }

    public String getName() {
        return name;
    }

    public MobEffectInstance[] getEffects() {
        return effects;
    }
}
